src/ICeWall/src/ICeWall.tcl

```diff
@@ -3561,26 +3904,1336 @@ namespace eval ICeWall {
+          }
+        }
+      } else {
+        if {[dict exists $library cells $bondpad_cell_ref orient]} {
+          if {[dict exists $library cells $bondpad_cell_ref orient $side_name]} {
+            dict set padcell bondpad orient [dict get $library cells $cell_ref orient $side_name]
+          } else {
+            if {[llength [dict get $library cells $bondpad_cell_ref orient]] == 1} {
+              dict set padcell bondpad orient [dict get $library cells $cell_ref orient]
+            } else {
+              utl::error PAD 136 "Unexpected value for orient attribute in library definition for $bondpad_cell_ref"
+            }
+          }
+        } else {
+          utl::error PAD 137 "No orientation information available for orientation of $cell_ref on side $side_name"
+        }
+      }
+      set bondpad_orient [dict get $padcell bondpad orient]
+      dict set padcell bondpad [verify_placement [dict get $padcell bondpad] $bondpad_cell_ref bondpad]
+    }
+    
+    # Verify bump 
+    if {[dict exists $padcell bump]} {
+      variable num_bumps_x
+      variable num_bumps_y
+
+      dict set padcell bump [check_bump [dict get $padcell bump]]
+
+      set row [dict get $padcell bump row]
+      set col [dict get $padcell bump col]
+    
+      if {$row < 1 || $row > $num_bumps_y} {
+        utl::error PAD 229 "The value for row is $row, but must be in the range 1 - $num_bumps_y"
+      }
+      if {$col < 1 || $col > $num_bumps_x} {
+        utl::error PAD 230 "The value for col is $col, but must be in the range 1 - $num_bumps_x"
+      }
+
+      set pitch [get_footprint_bump_pitch]
+      set origin [get_bump_origin $row $col]
+      
+      lassign [get_scaled_die_area] xMin yMin xMax yMax
+      switch {[dict get padcell side]} {
+        "bottom" {set xMin [expr [dict get $origin x] - $pitch / 2]]; set xMax [expr [dict get $origin x] + $pitch / 2]]}
+        "right"  {set yMin [expr [dict get $origin y] - $pitch / 2]]; set yMax [expr [dict get $origin y] + $pitch / 2]]}
+        "top"    {set xMin [expr [dict get $origin x] - $pitch / 2]]; set xMax [expr [dict get $origin x] + $pitch / 2]]}
+        "left"   {set yMin [expr [dict get $origin y] - $pitch / 2]]; set yMax [expr [dict get $origin y] + $pitch / 2]]}
+      }
+
+      if {[dict get $padcell cell scaled_centre x] < $xMin || [dict get $padcell cell scaled_centre x] > $xMax} {
+        utl::error PAD 163 "The padcell x location is [ord::dbu_to_microns [dict get $padcell cell scaled_centre x]], but for bump $row,$col to connect to a pad on the $side_name edge, $xMin <= x <= $xMax"
+      }
+      if {[dict get $padcell cell scaled_centre y] < $yMin || [dict get $padcell cell scaled_centre y] > $yMax} {
+        utl::error PAD 164 "The padcell y location is [ord::dbu_to_microns [dict get $padcell cell scaled_centre y]], but for bump $row,$col to connect to a pad on the $side_name edge, $yMin <= y <= $yMax"
```

Comment:
the  `$xMin <= x <= $xMax` and  `$yMin <= y <= $yMax` are reported in DBU and should be converted to microns to be consistent.

Reply:
Updated to report consistently in microns